File: freight-triage-polyglot-ledger-recovery/environment/native/src/rules/seal_and_lane_parity.cpp
#include "freight/rules.h"

namespace freight {

// Triage predicate: seal and lane parity.
bool rule_seal_and_lane_parity(const ProbeRecord& record) {
  return (record.sealLength + record.laneIndex) % 2 == 0;
}

}  // namespace freight

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/rules/SealAndLaneParity.java
package com.freight.rules;

/** Triage predicate: seal and lane parity. */
public final class SealAndLaneParity implements TriageRule {

    @Override
    public String name() {
        return "seal_and_lane_parity";
    }

    @Override
    public boolean apply(ProbeRecord record) {
        return (record.sealLength + record.laneIndex) % 2 == 0;
    }
}
